turbine/src/quic_endpoint.rs

```diff
@@ -168,27 +181,61 @@ async fn run_client(
     endpoint: Endpoint,
     mut receiver: AsyncReceiver<(SocketAddr, Bytes)>,
     sender: Sender<(Pubkey, SocketAddr, Bytes)>,
-    cache: Arc<RwLock<ConnectionCache>>,
+    router: Arc<AsyncRwLock<HashMap<SocketAddr, AsyncSender<Bytes>>>>,
+    cache: Arc<Mutex<HashMap<Pubkey, Connection>>>,
 ) {
     while let Some((remote_address, bytes)) = receiver.recv().await {
-        tokio::task::spawn(send_datagram_task(
+        let bytes = match router.read().await.get(&remote_address) {
+            None => bytes,
+            Some(sender) => match sender.try_send(bytes) {
+                Ok(()) => continue,
+                Err(TrySendError::Full(_)) => {
+                    error!("TrySendError::Full {remote_address}");
+                    continue;
+                }
+                Err(TrySendError::Closed(bytes)) => bytes,
+            },
+        };
+        let receiver = {
+            let mut router = router.write().await;
+            let bytes = match router.get(&remote_address) {
```

Comment:
nit: Was going to recommend trying to de-duplicate this block as it is nearly identical to the block above, but with the `continue` statement to control loop flow, I don't see a great way to do so unfortunately

Reply:
added a helper function to reduce the amount of duplicate code.